scripts/src/logging.ts

```diff
@@ -0,0 +1,72 @@
+import * as winston from "winston";
+import {GenericTextTransportOptions} from "winston";
+import {GenericTransportOptions} from "winston";
+import {FileTransportOptions} from "winston";
+
+export interface LogTarget {
+	name: string;
+	type: "console" | "file";
+	level: "debug" | "info" | "warn" | "error";
+	timestamp?: boolean | (() => string | boolean);
```

Comment:
I don't understand this. is timestamp a boolean?

Reply:
`timestamp` can be either a boolean or a function which returns either a boolean or a string.

Reply:
why is it a boolean? why not string or time in seconds from epoch?

Reply:
it is only to set whether or not a timestamp should be included.
if so, the timestamp is added.  it's not the format.